tests/ui/README.md

```diff
@@ -0,0 +1,64 @@
+## CodaLab UI Tests
+
+We test the CodaLab UI by using [Selenium](https://pypi.org/project/selenium/#history) and 
+[diffimg](https://pypi.org/project/diffimg/#history) for our screenshots comparison tool. Currently, our front end 
+automation only runs in Chrome and Firefox. We hope to support more browsers in the future. 
+
+### Setting up the tests on your local machine
+
+#### One-time setup
+
+1. If you haven't already, install the Python dependencies specified in `requirements.txt`, located in the root of the 
+repository, by running `pip install -r requirements.txt`.
+2. Next, install [ChromeDriver](https://chromedriver.chromium.org/getting-started) (for Chrome) and 
+[GeckoDriver](https://github.com/mozilla/geckodriver) (for Firefox). These drivers are the links between the selenium 
+tests and their respective browsers.
+
+#### Running the tests locally
+
+1. From the root of the repository, start the CodaLab service by running `./codalab_service.py start -bd`.
+2. Populate your local machine with a comprehensive sample worksheet by running 
+`docker exec -it codalab_rest-server_1 /bin/bash -c "python3 scripts/create_sample_worksheet.py"`. 
+3. Start the UI tests by running `python tests/ui/ui_tester.py`. You can also pass in an additional argument 
+`--headless`, which will run the tests in headless mode. 
+
+### Maintaining the baseline images
+
+If the screenshot tests are failing, check the `diff` folder for the diff images that are generated when image 
+comparisons fail. If the output images is as expected, accept them as the new baselines by replacing the stale baseline 
+images, located in `baselines`, with the newly generated images in `out`. If the failures are not expected, look into 
+your code changes, as they may have cause the tests to fail.
+
```

Comment:
Add a section about thresholds and how to change them if needed?

Reply:
The threshold is already at an upper bound of 9%. I don't really want people changing it. I will add a comment in the code.